### PR TITLE
FIX: Only do plain text table detection if outside code

### PR DESCRIPTION
**Pull request by venarius**
Plain text table detection now only happens outside of code fence blocks.

Resolves: https://meta.discourse.org/t/plain-text-table-detection-should-respect-code-fences/115894

### CODE REVIEW

**Comment by ZogStriP** [comment]
Nice, is there a way this can be tested?

File: test/javascripts/lib/pretty-text-test.js.es6 [hunk 1]
```diff
@@ -1387,6 +1387,13 @@ QUnit.test("enable/disable features", assert => {
   );
 });
 
+QUnit.test("table inside code block", assert => {
+  assert.cooked(
```
Comment (ZogStriP) [inline test/javascripts/lib/pretty-text-test.js.es6]: Is that testing your change? Isn't the fix related to _pasting_ a table in a code block?

Comment (venarius) [reply test/javascripts/lib/pretty-text-test.js.es6]: I think *real* pasting can't be tested with qunit, no?

Comment (venarius) [reply test/javascripts/lib/pretty-text-test.js.es6]: Also the problem here is that is has also worked before. We're just not wasting resources anymore now.

File: app/assets/javascripts/discourse/components/d-editor.js.es6 [hunk 2]
```diff
@@ -837,6 +828,13 @@ export default Ember.Component.extend({
 
         this.appEvents.trigger("composer:insert-text", markdown);
         handled = true;
+      } else if (plainText) {
```
Comment (vinothkannans) [inline app/assets/javascripts/discourse/components/d-editor.js.es6]: This is not intended behavior. I pushed the commit in https://github.com/discourse/discourse/commit/dc2ee78f7a667cd0ded8e198282def7ea51797bb.

**Comment by vinothkannans** [comment]
Done as per commit https://github.com/discourse/discourse/commit/dc2ee78f7a667cd0ded8e198282def7ea51797bb.

File: app/assets/javascripts/discourse/components/d-editor.js.es6 (hunk 2 above)
Comment (venarius) [reply app/assets/javascripts/discourse/components/d-editor.js.es6]: @vinothkannans can you tell me why this isn't intended behavior? It's always better if I know what exactly I did wrong.

Comment (vinothkannans) [reply app/assets/javascripts/discourse/components/d-editor.js.es6]: This `else if` condition will run only if ["HTML pasting"](https://github.com/venarius/discourse/blob/044a15afcb94e9a359fd95cb3ef82672cdf6c515/app/assets/javascripts/discourse/components/d-editor.js.es6#L823-L830) not happen. The intended behavior is table detection should happen even before HTML pasting. Only if pasting is not inside code block and HTML pasting enabled.

Comment (venarius) [reply app/assets/javascripts/discourse/components/d-editor.js.es6]: Oh okay, got it! Thanks!

**Pull request closed without merging**